### PR TITLE
2: Converted to a multi-module project. Added generated OB data model

### PR DESCRIPTION
**Pull request by matt-wills**
Converted to a multi-module project
Added OB model objects (including the ability to generate them) to `forgerock-sba-ob-model`

**Issue:** https://github.com/SecureBankingAcceleratorToolkit/SecureBankingAcceleratorToolkit/issues/2

### CODE REVIEW

File: forgerock-sba-ob-aspsp-mock/pom.xml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                <artifactId>maven-failsafe-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-maven-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
+
+    <scm>
```
Comment (jorgesanchezperez) [inline forgerock-sba-ob-aspsp-mock/pom.xml:116]: The repository name must be fixed.

**Review by jorgesanchezperez — DISMISSED** [review]

File: forgerock-sba-ob-aspsp-mock/pom.xml (hunk 1 above)
Comment (matt-wills) [reply forgerock-sba-ob-aspsp-mock/pom.xml:116]: Oops, I corrected it in the project's parent pom earlier, but forgot I left it in here. I've removed it from this file entirely.

**Review by jorgesanchezperez — APPROVED** [review]

**Review by jorgesanchezperez — APPROVED** [review]

**Review by jorgesanchezperez — APPROVED** [review]

**Comment by BohoCode** [comment]
What's the (#2) #3 stuff in the title of this PR? Confusing?

File: pom.xml [hunk 2]
```diff
@@ -25,33 +25,36 @@
...
+    <modules>
+        <module>securebanking-openbanking-aspsp-model</module>
+        <module>securebanking-openbanking-aspsp-mock</module>
+    </modules>
+
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>2.3.4.RELEASE</version>
+        <version>2.3.5.RELEASE</version>
```
Comment (BohoCode) [inline pom.xml:44]: Use the securebanking-parent? Can do that in another PR

**Comment by matt-wills** [comment]
> What's the (#2) #3 stuff in the title of this PR? Confusing?

When I filled in the description whilst creating the PR, I put (#2) on the end to indicate which issue it refers to. Github then appends the PR number to the end! I guess I should just put a link to #2 in the main description somewhere?

**Comment by BohoCode** [comment]
Interactive rebase down to one commit please

**Comment by matt-wills** [comment]
> Interactive rebase down to one commit please

Done.

**Review by jorgesanchezperez — APPROVED** [review]

**Review by BohoCode — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (matt-wills) [reply pom.xml:44]: I'll move a few libraries to the parent pom in a separate PR 👍🏻
